WordPress/openverse
# Remove `proxy` service and make API NGINX setup similar to the frontend

## Fixes



Partially addresses #4108 by @dhruvkb

## Description



This PR
- removes the service `proxy` which was meant to test the API running under HTTPS locally. This was rarely or maybe even never used.
  - also removes the documentation and scaffolding files related to this service
- makes the API NGINX configuration identical to the frontend
  - also uses the `snippets/` pattern used in the frontend for common config in API locations
- updates Docker Compose profiles and `just` recipes so that the `api_extra` profile depends on the `api` profile

## Testing Instructions


Check out this PR and verify that the `nginx` service builds and runs as expected.

```bash
just build nginx && just api/up-extra && just logs nginx
```

## Checklist


      PRs) or the media properties generator (`./ov just catalog/generate-docs media-props`
      for the catalog or `./ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
